data/impl/RecordSet.js

```diff
@@ -127,6 +127,29 @@ export class RecordSet {
 
         newRecords.forEach((newRecord, id) => {
             const currRecord = existingRecords.get(id);
+
+            if (currRecord) {
+                throwIf(newRecord.parent && newRecord.parent.id !== newRecord.parent.id,
+                    'Cannot change parents via updateData!');
+                newRecord.parent = currRecord.parent;
+            }
+
+            let hasChildren = false;
+            for (let rec of existingRecords.values()) {
```

Comment:
Is there a better way to do this with a Map? `forEach` might be cleaner but also cannot early-out..

Reply:
Is the goal here to detect if there are pre-existing record(s) that think the new record being evaluated is their parent, and remove them if they aren't in the new set? If so, can we look at `currRecord.children`? 

Some tricky logic here, although I'll be honest i am in transit and it's not an ideal place for focusing.  But maybe that compromised state is a useful way to detect where an extra comment or two would help. 😄 In particular the interplay between existing records vs. new ones is difficult for me to keep straight - what side we're looking to clean up and why/when.

Reply:
I need to dig back into this but I believe I didn't want to call `currRecord.children` because that would create the children map (if not already created), and didn't want to do that if there was a way to detect this without it, but need to find the balance between performance and readability here.

Will add some comments explaining why we have/need this logic